### PR TITLE
Clarify docs about rank aggregate

### PR DESCRIPTION
**Pull request by shati-patel**
The `rank` aggregate can be pretty confusing, so here are two small changes: 
- Mention that `rank[1](...)` is the same as `min(...)`
- Make this (+ note about 1-based indexing) more visible by pulling them into a separate **note** block

(Second commit is an unrelated change to fix a table that was formatted incorrectly.)

### CODE REVIEW

File: docs/codeql/ql-language-reference/expressions.rst [hunk 1]
```diff
@@ -308,7 +308,10 @@ The following aggregates are available in QL:
 
       rank[4](int i | i = [5 .. 15] | i)
 
-  Note that the rank indices start at ``1``, so ``rank[0](...)`` returns no results.
+  .. pull-quote:: Note
+
+     - Rank indices start at ``1``, so ``rank[0](...)`` returns no results.
+     - ``rank[1](...)`` is the same as ``min(...)``.
```
Comment (MathiasVP) [inline docs/codeql/ql-language-reference/expressions.rst:314]: This is a bit unrelated, but anyway: I remember reading in https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md that QLDoc shouldn't use the word `return`. Should we maybe replace "returns no result" with "has no results"?

Comment (shati-patel) [reply docs/codeql/ql-language-reference/expressions.rst]: Oh nice, thanks! I've updated it here. Other areas of docs probably use this incorrectly too, but checking those can be a separate job 😅

**Review by dbartol — APPROVED** [review]
Definitely helpful!

**Review by geoffw0 — APPROVED** [review]
LGTM.

**Pull request merged**